src/Polly.Core/Simmy/Outcomes/OutcomeChaosStrategy.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Polly.Telemetry;
+
+namespace Polly.Simmy.Outcomes;
+
+#pragma warning disable S3928 // Custom ArgumentNullException message
+
+internal class OutcomeChaosStrategy<T> : MonkeyStrategy<T>
```

Comment:
I suggest to actually split fault strategy from outcome:

``` csharp
internal class OutcomeChaosStrategy : MonkeyStrategy
{
    private readonly ResilienceStrategyTelemetry _telemetry;

    public OutcomeChaosStrategy(FaultStrategyOptions options, ResilienceStrategyTelemetry telemetry)
        : base(options)
    {
        if (options.Fault is null && options.FaultGenerator is null)
        {
            throw new ArgumentNullException(nameof(options.Fault), "Either Fault or FaultGenerator is required.");
        }

        _telemetry = telemetry;
        Fault = options.Fault;
        OnFaultInjected = options.OnFaultInjected;
        FaultGenerator = options.FaultGenerator is not null ? options.FaultGenerator : (_) => new(options.Fault);
    }

    public Func<OnFaultInjectedArguments, ValueTask>? OnFaultInjected { get; }

    public Func<FaultGeneratorArguments, ValueTask<Exception?>>? FaultGenerator { get; }

    public Exception? Fault { get; }

    protected internal async override ValueTask<Outcome<TResult>> ExecuteCore<TResult, TState>(
        Func<ResilienceContext, TState, ValueTask<Outcome<TResult>>> callback,
        ResilienceContext context,
        TState state)
    {
        try
        {
            if (await ShouldInjectAsync(context).ConfigureAwait(context.ContinueOnCapturedContext))
            {
                var fault = await InjectFault(context).ConfigureAwait(context.ContinueOnCapturedContext);
                if (fault is not null)
                {
                    return new Outcome<TResult>(fault);
                }
            }

            return await StrategyHelper.ExecuteCallbackSafeAsync(callback, context, state).ConfigureAwait(context.ContinueOnCapturedContext);
        }
        catch (OperationCanceledException e)
        {
            return new Outcome<TResult>(e);
        }
    }

    private async ValueTask<Exception?> InjectFault(ResilienceContext context)
    {
        var fault = await FaultGenerator!(new(context)).ConfigureAwait(context.ContinueOnCapturedContext);
        if (fault is null)
        {
            return null;
        }

        var args = new OnFaultInjectedArguments(context, fault);
        _telemetry.Report(new(ResilienceEventSeverity.Information, OutcomeConstants.OnFaultInjectedEvent), context, args);

        if (OnFaultInjected is not null)
        {
            await OnFaultInjected(args).ConfigureAwait(context.ContinueOnCapturedContext);
        }

        return fault;
    }
}
```

The strategy would be cleaner too.

Reply:
Yeah I also thought about it and I think it's a good idea! Let's merge this guy first so that I don't keep slowing down the release and I'll take care of this and [this](https://github.com/App-vNext/Polly/pull/1571#discussion_r1319409359) next weekend.